src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -879,28 +869,28 @@ export class SearchSource {
         // inject the format from the computed fields if one isn't given
         const docvaluesIndex = keyBy(filteredDocvalueFields, 'field');
         const bodyFields = this.getFieldsWithoutSourceFilters(index, body.fields);
-        body.fields = uniqWith(
-          bodyFields.concat(filteredDocvalueFields),
-          (fld1: SearchFieldValue, fld2: SearchFieldValue) => {
-            const field1Name = this.getFieldName(fld1);
-            const field2Name = this.getFieldName(fld2);
-            return field1Name === field2Name;
+
+        const uniqueFieldNames = new Set();
+        const uniqueFields = [];
+        for (const field of bodyFields.concat(filteredDocvalueFields)) {
```

Comment:
We could also change to `forEach` here instead of `for` to minimize event loop blocking.

Reply:
thx, you mean? 

` bodyFields.concat(filteredDocvalueFields).forEach` right?  wouldn't this also block the event loop?

Reply:
Right. Well, it will block in chunks but it might also allow some browser work in between. Or maybe I am just making this up 😀

Reply:
I think it would be the case if it was an async/await type of forEach, but here I think it was just block the even look until done (pls prove me wrong 😃 )